### ISSUE TITLE
CellML Text view: an id on a MathML element should be serialised using the MathML id attribute rather than cmeta:id

### ISSUE THREAD

**Issue by mirams** [opencor/opencor#658]
After editing a CellML file in 0.4.1 I got this error in Chaste:

    /home/garmir/workspace/Chaste/projects/ApPredict/src/cellml/luo_rudy_1994.cellml:86:0:ERROR:RELAXNGV:RELAXNG_ERR_EXTRACONTENT: Element math has extra content: apply
    <string>:0:0:ERROR:RELAXNGV:RELAXNG_ERR_INTEREXTRA: Extra element math in interleave
    /home/garmir/workspace/Chaste/projects/ApPredict/src/cellml/luo_rudy_1994.cellml:85:0:ERROR:RELAXNGV:RELAXNG_ERR_CONTENTVALID: Element component failed to validate content
    <string>:0:0:ERROR:RELAXNGV:RELAXNG_ERR_INTEREXTRA: Extra element component in interleave
    /home/garmir/workspace/Chaste/projects/ApPredict/src/cellml/luo_rudy_1994.cellml:65:0:ERROR:RELAXNGV:RELAXNG_ERR_CONTENTVALID: Element model failed to validate content
    /home/garmir/workspace/Chaste/projects/ApPredict/src/cellml/luo_rudy_1994.cellml is not a valid CellML file
    scons: *** [projects/ApPredict/build/debug/src/cellml/luo_rudy_1994.cpp] Error 1
    scons: building terminated because of errors.

all I did was multiply a number in LR94 by a dimensionless constant in one calculation, save, then attempt to load in Chaste.

**Comment by agarny** [comment]
It would be good to have both the original and your modified version of the LR94 model, so that we can compare them and see exactly what has been added by the CellML Text view (I imagine it is the one you used?) and whether it is valid CellML.

**Comment by mirams** [comment]
So it seems that Chaste doesn't like

    <math xmlns="http://www.w3.org/1998/Math/MathML">
	<apply cmeta:id="dV_dt_eq">

but is fine with

    <math xmlns="http://www.w3.org/1998/Math/MathML">
	<apply id="dV_dt_eq">

So question is whether a cmeta:id inside MathML makes sense or not?

**Comment by agarny** [comment]
Yes, I saw that on Chaste's [tracker item #2692](https://chaste.cs.ox.ac.uk/trac/ticket/2692). I must confess that I didn't remember about MathML having its own ```id``` attribute and OpenCOR certainly doesn't do anything special when wanting to set an id for a ```<mathml:apply>``` element.

Just to be sure, I checked the CellML specifications and all I could find (on a quick search) is the following paragraph at the end of section 4.3 (which is about some CellML example):

> The ```<mathml:apply>``` element at the top level of the expression defines an id attribute, which can be used to associate further metadata with the expression. The linking of metadata with elements in a CellML document is described in more detail in Section 8.2.

So, it looks like the MathML ```id``` attribute has to be used instead of ```cmeta:id```. Fair enough.

I have come up with a fix for this, which I am going to push later on. From there, I think I will generate a new snapshot within the next few days.
